src/org/loklak/LoklakServer.java

```diff
@@ -167,49 +173,200 @@ public static void main(String[] args) throws Exception {
-/*
-        Server server = new Server();
-        ServerConnector connector = new ServerConnector(server);
-        connector.setPort(9999);
-         
-        HttpConfiguration https = new HttpConfiguration();
-        https.addCustomizer(new SecureRequestCustomizer());
-         
-        SslContextFactory sslContextFactory = new SslContextFactory();
-        File keystore = new File(DAO.conf_dir, DAO.getConfig("keystore.name", "keystore.jks"));
-        sslContextFactory.setKeyStorePath(keystore.getAbsolutePath());
-        sslContextFactory.setKeyStorePassword(DAO.getConfig("keystore.password", ""));
-        sslContextFactory.setKeyManagerPassword(DAO.getConfig("keystore.password", ""));
-        
-        ServerConnector sslConnector = new ServerConnector(server,
-                new SslConnectionFactory(sslContextFactory, "http/1.1"),
-                new HttpConnectionFactory(https));
-        sslConnector.setPort(9998);
-        
-        server.setConnectors(new Connector[] { connector, sslConnector });
-*/
-        /// https
-        
         // init the http server
         QueuedThreadPool pool = new QueuedThreadPool();
         pool.setMaxThreads(500);
         LoklakServer.server = new Server(pool);
         LoklakServer.server.setStopAtShutdown(true);
+        
+        //http
+        HttpConfiguration http_config = new HttpConfiguration();
+        if(redirect) {
+        	http_config.addCustomizer(new SecureRequestCustomizer());
+        	http_config.setSecureScheme("https");
+        	http_config.setSecurePort(httpsPort);
+        }
+        
         ServerConnector connector = new ServerConnector(LoklakServer.server);
+        connector.addConnectionFactory(new HttpConnectionFactory(http_config));
         connector.setPort(httpPort);
         connector.setName("httpd:" + httpPort);
         connector.setIdleTimeout(20000); // timout in ms when no bytes send / received
         LoklakServer.server.addConnector(connector);
+        
+        //https
+        //keytool -genkey -alias sitename -keyalg RSA -keystore keystore.jks -keysize 2048
+        //uncommented lines for http2 (jetty 9.3 / java 8)        
+        if(useHttps){
+        	
+        	String keySource = DAO.getConfig("https.keysource", "keystore");
+        	String keystorePath = null;
+        	String keystorePass = null;
+        	String keystoreManagerPass = null;
+        	
+        	if(keySource.equals("keystore")){
+        		File keystore = new File(DAO.conf_dir, DAO.getConfig("keystore.name", "keystore.jks"));
+        		if(!keystore.exists() || !keystore.isFile() || !keystore.canRead()){
+        			Log.getLog().info("Could not find keystore");
+    	            System.exit(-1);
+        		}
+        		keystorePath = keystore.getAbsolutePath();
+        		keystorePass = DAO.getConfig("keystore.password", "");
+        		keystoreManagerPass = DAO.getConfig("keystore.password", "");
+        	}
+        	else if (keySource.equals("key-cert")){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2550015915/issues/source?bid=3288751&fileBranchId=3288809#l248)